swiftsimio/initial_conditions/generate_particles.py

```diff
@@ -1,5 +1,1304 @@
+                        # compute displacement for overdense particle
+                        dx = np.zeros(3, dtype=np.float)
+                        H = kernel_gamma * h[uind]
+                        for d in range(ndim):
+                            sign = 1 if np.random.random() < 0.5 else -1
+                            dx[d] = np.random.uniform() * 0.3 * H * sign
+
+                        x[oind] = x[uind] + dx
+                        touched[oind] = True
+                        touched[uind] = True
+                        moved[nmoved] = oind
+                        nmoved += 1
+                        break
+
+        print("Moved:", nmoved)
+        if nmoved > 0:
+
+            # check boundary conditions
+            if self.periodic:
+                for d in range(ndim):
+                    x[x[:, d] > boxsize[d], d] -= boxsize[d]
+                    x[x[:, d] < 0.0, d] += boxsize[d]
+            else:
+                temp = 1.0 / npart ** (1.0 / ndim)
+                for d in range(ndim):
+
+                    # move them away from the edge by a random factor of mean "cell size" boxsize/npart^ndim
+                    mask = x[:, d] > boxsize[d]
+                    x[mask, d] = boxsize[d] * (
+                        1.0 - np.random.uniform(x[mask].shape) * temp
+                    )
+
+                    mask = x[:, d] < 0.0
+                    x[mask, d] = np.random.uniform(x[mask, d].shape) * boxsize[d] * temp
+
+        return moved[:nmoved]
+
+    def dump_current_state(
+        self, iteration: int, h: np.ndarray, rho: np.ndarray,
+    ):
+        r"""
+        Write an intermediate output of current particle positions, densities,
+        masses, and smoothing lengths.
+        Use the default swift IC format for the dump, so invent meaningless
+        values for required quantities ``velocity`` and ``internal_energy``.
+
+        Parameters
+        ----------------
+
+        iteration: int
+            current iteration number
+
+        h: np.ndarray
+            particle smoothing lengths
+
+        rho: np.ndarray
+            particle densities
+
+        """
+        ndim = self.ndim
+        npart = self.npart
+        x = self.x
+        m = self.m
+
+        u_l = self.unitsys["length"]
+        u_m = self.unitsys["mass"]
+        u_t = self.unitsys["time"]
+
+        W = Writer(self.unitsys, self.boxsize)
+        W.gas.coordinates = unyt.unyt_array(x, u_l)
+        W.gas.smoothing_length = unyt.unyt_array(h, u_l)
+        W.gas.masses = unyt.unyt_array(m, u_m)
+        W.gas.densities = unyt.unyt_array(rho, u_m / u_l ** ndim)
+
+        # invent some junk to fill up necessary arrays
+        W.gas.internal_energy = unyt.unyt_array(
+            np.ones(npart, dtype=np.float), u_l ** 2 / u_t ** 2
+        )
+        W.gas.velocities = unyt.unyt_array(np.zeros(npart, dtype=np.float), u_l / u_t)
+
+        fname = self.runparams.dump_basename + str(iteration).zfill(5) + ".hdf5"
```

Comment:
I would prefer if we could use f-string formatting here and 4-filled zeros to be consistent with the rest of `swiftsimio` and SWIFT itself:

```
filename = f"{self.runparams.dump_basename}_{iteration:04d}.hdf5"
```

Reply:
I wouldn't. I had plenty of cases where I ended up working with 10000+ iterations.

Reply:
Changed to f-string formatting. Keeping the 5 digits.